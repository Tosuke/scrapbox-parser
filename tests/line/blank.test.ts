/* global describe it expect */
/* eslint-disable no-irregular-whitespace */

describe('blank', () => {
  it('Simple half-space blank', () => {
    expect('[ ]').toMatchSnapshotWhenParsing()
  })

  it('Simple double-byte space blank', () => {
    expect('[　]').toMatchSnapshotWhenParsing()
  })

  it('Simple tab blank', () => {
    expect('[\t]').toMatchSnapshotWhenParsing()
  })

  it('Multi char blank', () => {
    expect('[ 　 \t　\t ]').toMatchSnapshotWhenParsing()
  })

  it('Blank in the sentence', () => {
    expect('sentence[ ]sentence').toMatchSnapshotWhenParsing()
  })

  it('[] is not blank', () => {
    expect('[]').toMatchSnapshotWhenParsing()
  })

  it('Blank in the [*** ]', () => {
    expect('[*** [ ]]').toMatchSnapshotWhenParsing()
  })
})
