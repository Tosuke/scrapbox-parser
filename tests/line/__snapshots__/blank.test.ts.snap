// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blank [] is not blank: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "[]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Blank in the [*** ]: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "[*** ",
        "type": "plain",
      },
      Object {
        "text": " ",
        "type": "blank",
      },
      Object {
        "text": "]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Blank in the sentence: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "sentence",
        "type": "plain",
      },
      Object {
        "text": " ",
        "type": "blank",
      },
      Object {
        "text": "sentence",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Multi char blank: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": " 　 	　	 ",
        "type": "blank",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Simple double-byte space blank: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "　",
        "type": "blank",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Simple half-space blank: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": " ",
        "type": "blank",
      },
    ],
    "type": "line",
  },
]
`;

exports[`blank Simple tab blank: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "text": "	",
        "type": "blank",
      },
    ],
    "type": "line",
  },
]
`;
